sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -303,6 +330,10 @@ private[orc] object OrcRelation extends HiveInspectors {
     maybeStructOI.map(unwrap).getOrElse(Iterator.empty)
   }
 
+  def isMismatchSchema(physicalSchema: StructType, requestedSchema: StructType): Boolean = {
+    requestedSchema.forall(a => physicalSchema.getFieldIndex(a.name).isEmpty)
```

Comment:
Oh, never mind. It's really about checking whether all requested columns are not matched.